.github/workflows/apps/outdated_github_majors.go

```diff
@@ -0,0 +1,245 @@
+// 	// return the largest major version associated
+
+// GetLatestMajorContrib finds the latest major version of a repository in the contrib directory.
+func GetLatestMajorContrib(repository string) (string, error) {
+	const contribDir = "contrib"
+
+	// Prepare the repository matching pattern
+	// ex. repository = "redis/go-redis" should match on any directory that contains redis/go-redis
+	repoPattern := fmt.Sprintf(`^%s/%s(/.*)?`, contribDir, strings.ReplaceAll(repository, "/", "/"))
+
+	// Compile regex for matching contrib directory entries
+	repoRegex, err := regexp.Compile(repoPattern)
+	if err != nil {
+		return "", fmt.Errorf("invalid repository pattern: %v", err)
+	}
+
+	var latestVersion string
+
+	// Walk through the contrib directory
+	err = filepath.Walk(contribDir, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Check if the path matches the contrib repository pattern
+		if !info.IsDir() || !repoRegex.MatchString(path) {
+			return nil
+		}
+
+		// Look for go.mod file
+		goModPath := filepath.Join(path, "go.mod")
+		if _, err := os.Stat(goModPath); err != nil {
+			return nil // No go.mod file, skip
+		}
+
+		// Parse the go.mod file to find the version
+		version, err := extractVersionFromGoMod(goModPath, repository)
+		if err != nil {
+			return nil
+		}
+
+		// Update the latest version if it's greater
+		if latestVersion == "" || semver.Compare(version, latestVersion) > 0 {
+			latestVersion = version
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return "", fmt.Errorf("error walking contrib directory: %v", err)
+	}
+
+	if latestVersion == "" {
+		return "", errors.New("no matching contrib repository with a valid version found")
+	}
+
+	return latestVersion, nil
+}
+
+func extractVersionFromGoMod(goModPath string, repository string) (string, error) {
+	data, err := os.ReadFile(goModPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to read go.mod: %v", err)
+	}
+	var versions []string
+
+	// parse go.mod
+	modFile, err := modfile.Parse(goModPath, data, nil)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse go.mod: %v", err)
+	}
+
+	for _, req := range modFile.Require {
+		if strings.Contains(req.Mod.Path, repository) { // check if module path contains repository substring
+			if semver.IsValid(req.Mod.Version) {
+				versions = append(versions, req.Mod.Version)
+			}
+		}
+
+	}
+
+	if len(versions) == 0 {
+		return "", errors.New("no valid versions found for repository in go.mod")
+	}
+
+	// Sort versions in descending order
+	// Return the greatest version
+	sort.Slice(versions, func(i, j int) bool {
+		return semver.Compare(versions[i], versions[j]) > 0
+	})
+
+	return versions[0], nil
+
+}
+
+// ValidateRepository checks if the repository string starts with "github.com" and ends with a version suffix.
+// Returns true if the repository is valid, false otherwise.
+func ValidateRepository(repo string) bool {
```

Comment:
nit: as this is a CLI tool, you don't need to export functions. Usually, in these cases, you would do `validateRepository` and the same for the other functions that start with uppercase.